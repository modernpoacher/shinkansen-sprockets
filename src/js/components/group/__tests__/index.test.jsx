import React from 'react'
import renderer from 'react-test-renderer'

import Group from '#sprockets/components/group'

describe('#sprockets/components/group', () => {
  describe('<Group />', () => {
    describe('With required props', () => {
      const component = (
        <Group />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('`getClassName`', () => {
        it('is defined', () => {
          return expect(Group.prototype.getClassName)
            .toBeDefined()
        })
      })

      describe('`shouldComponentUpdate`', () => {
        it('is defined', () => {
          return expect(Group.prototype.shouldComponentUpdate)
            .toBeDefined()
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Group>
            MOCK CHILDREN
          </Group>
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      it('returns the classname', () => {
        const component = (
          <Group />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        return expect(instance.getClassName())
          .toBe('group')
      })
    })

    describe('`shouldComponentUpdate()`', () => {
      const component = (
        <Group>
          MOCK CHILDREN
        </Group>
      )

      let instance

      beforeEach(() => {
        instance = (
          renderer.create(component)
            .getInstance()
        )
      })

      describe('`props` have changed', () => {
        it('returns true', () => {
          return expect(instance.shouldComponentUpdate({
            children: null
          }))
            .toBe(true)
        })
      })

      describe('`props` have not changed', () => {
        it('returns false', () => {
          return expect(instance.shouldComponentUpdate({ // instance.props
            children: 'MOCK CHILDREN'
          }))
            .toBe(false)
        })
      })
    })
  })
})
