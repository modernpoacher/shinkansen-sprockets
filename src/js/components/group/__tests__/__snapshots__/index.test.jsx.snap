// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#sprockets/components/group <Group /> With additional props renders 1`] = `
<fieldset
  className="group"
>
  MOCK CHILDREN
</fieldset>
`;

exports[`#sprockets/components/group <Group /> With required props renders 1`] = `
<fieldset
  className="group"
/>
`;
